src/RestControllers/FHIR/FhirPatientRestController.php

```diff
@@ -96,17 +96,28 @@ public function getAll($search)
         }
 
         $searchParam = array(
-            'name' => $search['name'],
             'DOB' => $search['birthdate'],
             'city' => $search['address-city'],
             'state' => $search['address-state'],
             'postal_code' => $search['address-postalcode'],
             'phone_contact' => $search['phone'],
-            'address' => $search['address'],
             'sex' => $search['gender'],
+            'lname' => $search['name'],
+            'fname' => $search['name'],
+            'city' => $search['address'],
+            'street' => $search['address'],
+            'state' => $search['address'],
+            'postal_code' => $search['address'],
```

Comment:
hi @yashrajbothra , Seems really odd to have duplicate keys of `city`, `state`, and `postal_code` in this array?

Reply:
Whoops, I think I missed that point thanks for pointing it out. I think if we went this way we can search either one of the elements like we can search `address` or any of its components we can't search for both as there is only one key in a database. I think the better idea will be to make a helper function which can do some magic whenever a key does not exist in our database 😄